src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -262,13 +262,17 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
     else:
         option = DiskCreateOption.empty
 
+    if upload_size_byte is not None and for_upload is not True:
```

Comment:
When `for_upload` is None, should we default to `True`?

Reply:
When `for_upload` is None, it means this disk is not created for upload.